x-pack/plugins/ml/public/application/timeseriesexplorer/timeseriesexplorer.js

```diff
@@ -463,9 +463,9 @@ export class TimeSeriesExplorer extends React.Component {
               if (
                 // If the user's focus range is not defined (i.e. no 'zoom' parameter restored from the appState URL),
                 // then calculate the default focus range to use
-                zoom === undefined &&
-                (focusRange === undefined ||
-                  this.previousSelectedForecastId !== this.props.selectedForecastId)
+                zoom === undefined ||
```

Comment:
This is fixing the bug when coming in from an annotation, but the issue fixed by https://github.com/elastic/kibana/pull/176969 is happening again where the zoom is not restored if the URL contains a `forecastId` and `zoom` parameters.

Reply:
Confirmed this is fixed by https://github.com/elastic/kibana/pull/182176/commits/d614c66d3bcac75ff9db89131ee71842ee43e475